react/src/components/ShellScriptEditModal.tsx

```diff
@@ -0,0 +1,321 @@
+      }
+      onCancel={() => onRequestClose()}
+      okText={t('button.Save')}
+      footer={[
+        <Flex justify="between" style={{ width: '100%' }}>
+          <Flex>
+            <Dropdown.Button
+              type="default"
+              danger
+              style={{ width: 'fit-content' }}
+              menu={{
+                items: [
+                  {
+                    key: 'reset',
+                    label: t('button.Reset'),
+                    onClick: () => {
+                      app.modal.confirm({
+                        title: t('dialog.title.LetsDouble-Check'),
+                        content: t('dialog.ask.DoYouWantToResetChanges'),
+                        onOk: () => {
+                          if (shellInfo === 'bootstrap') {
+                            setScript('');
+                          }
+                          if (shellInfo === 'userconfig') {
+                            setScript('');
+                          }
+                        },
+                      });
+                    },
+                    danger: true,
+                  },
+                ],
+              }}
+              onClick={() => {
+                app.modal.confirm({
+                  title: t('dialog.title.LetsDouble-Check'),
+                  content: t('dialog.ask.DoYouWantToDeleteSomething', {
+                    name:
+                      shellInfo === 'bootstrap'
+                        ? 'Bootstrap script'
+                        : `${rcfileNames}`,
+                  }),
+                  onOk: deleteScript,
+                });
+              }}
+            >
+              <DeleteOutlined />
+            </Dropdown.Button>
+          </Flex>
+          <Flex gap={'sm'}>
+            <Button
+              key="cancel"
+              onClick={() => onRequestClose()}
+              style={{ width: 'fit-content' }}
+            >
+              {t('button.Cancel')}
+            </Button>
+            <Dropdown.Button
+              key="submit"
+              type="primary"
+              onClick={() => {
+                saveScript();
+              }}
+              style={{ width: 'fit-content' }}
+              menu={{
+                items: [
+                  {
+                    key: 'save',
+                    label: t('button.SaveWithoutClose'),
+                    onClick: () => saveScript({ closeAfter: false }),
+                  },
+                ],
+              }}
+            >
+              {t('button.SaveAndClose')}
+            </Dropdown.Button>
+          </Flex>
+        </Flex>,
+      ]}
+      destroyOnClose
+      {...modalProps}
+    >
+      <Flex direction="column" align="stretch" gap={'sm'}>
+        <Typography.Text>
+          {t('usersettings.BootstrapScriptDescription')}
+        </Typography.Text>
+        {shellInfo === 'userconfig' && (
+          <Select
+            onClear={() => setRcfileNames('.bashrc')}
```

Comment:
Is there any reason to set onClear even if `allowClear` is not setted?

Reply:
I've deleted it.